src/osd/OSDMap.cc

```diff
@@ -1394,150 +1376,126 @@ int OSDMap::apply_incremental(const Incremental &inc)
+    pools[elem.first] = elem.second;
+    pools[elem.first].last_change = epoch;
   }
-  for (map<int64_t,string>::const_iterator p = inc.new_pool_names.begin();
-       p != inc.new_pool_names.end();
-       ++p) {
-    auto pool_name_entry = pool_name.find(p->first);
+
+  for (const auto &elem : inc.new_pool_names) {
+    auto pool_name_entry = pool_name.find(elem.first);
     if (pool_name_entry != pool_name.end()) {
       name_pool.erase(pool_name_entry->second);
-      pool_name_entry->second = p->second;
+      pool_name_entry->second = elem.second;
     } else {
-      pool_name[p->first] = p->second;
+      pool_name[elem.first] = elem.second;
     }
-    name_pool[p->second] = p->first;
+    name_pool[elem.second] = elem.first;
   }
-  for (set<int64_t>::const_iterator p = inc.old_pools.begin();
-       p != inc.old_pools.end();
-       ++p) {
-    pools.erase(*p);
-    name_pool.erase(pool_name[*p]);
-    pool_name.erase(*p);
+  for (const auto &elem : inc.old_pools) {
+    pools.erase(elem);
+    name_pool.erase(pool_name[elem]);
+    pool_name.erase(elem);
   }
 
-  for (map<int32_t,uint32_t>::const_iterator i = inc.new_weight.begin();
-       i != inc.new_weight.end();
-       ++i) {
-    set_weight(i->first, i->second);
+  for (const auto &elem : inc.new_weight) {
+    set_weight(elem.first, elem.second);
 
     // if we are marking in, clear the AUTOOUT and NEW bits, and clear
     // xinfo old_weight.
-    if (i->second) {
-      osd_state[i->first] &= ~(CEPH_OSD_AUTOOUT | CEPH_OSD_NEW);
-      osd_xinfo[i->first].old_weight = 0;
+    if (elem.second) {
+      osd_state[elem.first] &= ~(CEPH_OSD_AUTOOUT | CEPH_OSD_NEW);
+      osd_xinfo[elem.first].old_weight = 0;
     }
   }
 
-  for (map<int32_t,uint32_t>::const_iterator i = inc.new_primary_affinity.begin();
-       i != inc.new_primary_affinity.end();
-       ++i) {
-    set_primary_affinity(i->first, i->second);
+  for (const auto &elem : inc.new_primary_affinity) {
+    set_primary_affinity(elem.first, elem.second);
   }
 
   // erasure_code_profiles
-  for (vector<string>::const_iterator i = inc.old_erasure_code_profiles.begin();
-       i != inc.old_erasure_code_profiles.end();
-       ++i)
-    erasure_code_profiles.erase(*i);
+  for (const auto &elem : inc.old_erasure_code_profiles)
+    erasure_code_profiles.erase(elem);
   
-  for (map<string,map<string,string> >::const_iterator i =
-	 inc.new_erasure_code_profiles.begin();
-       i != inc.new_erasure_code_profiles.end();
-       ++i) {
-    set_erasure_code_profile(i->first, i->second);
+  for (const auto &elem : inc.new_erasure_code_profiles) {
+    set_erasure_code_profile(elem.first, elem.second);
   }
   
   // up/down
-  for (map<int32_t,uint8_t>::const_iterator i = inc.new_state.begin();
-       i != inc.new_state.end();
-       ++i) {
-    int s = i->second ? i->second : CEPH_OSD_UP;
-    if ((osd_state[i->first] & CEPH_OSD_UP) &&
+  for (const auto &elem : inc.new_state) {
```

Comment:
s/elem/state/ or better off define an const local variable to reference `elem.first`, like
```c++
const auto osd = state.first;
```